이승창/RestApi/src/main/java/com/example/restapi/global/dto/ApiResponseTemplate.java

```diff
@@ -0,0 +1,42 @@
+package com.example.restapi.global.dto;
+
+import com.example.restapi.global.exception.code.ErrorCode;
+import com.example.restapi.global.exception.code.SuccessCode;
+import lombok.*;
+import org.springframework.http.ResponseEntity;
+
+@Getter
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Builder
+public class ApiResponseTemplate<T> {
+
+    private final int status;       // HTTP 상태 코드
+    private final boolean success;  // 성공 여부
+    private final String message;   // 메시지
+    private T data;                 // 실제 응답 데이터
+
+    // 성공 응답
+    public static <T> ResponseEntity<ApiResponseTemplate<T>> success(SuccessCode successCode, T data) {
+        return ResponseEntity
+                .status(successCode.getHttpStatus())
+                .body(ApiResponseTemplate.<T>builder()
+                        .status(successCode.getHttpStatus().value())
+                        .success(true)
+                        .message(successCode.getMessage())
+                        .data(data)
+                        .build());
+    }
+
+    // 실패 응답
+    public static <T> ResponseEntity<ApiResponseTemplate<T>> error(ErrorCode errorCode, T data) {
+        return ResponseEntity
+                .status(errorCode.getHttpStatus())
+                .body(ApiResponseTemplate.<T>builder()
+                        .status(errorCode.getHttpStatus().value())
+                        .success(false)
+                        .message(errorCode.getMessage())
+                        .data(data)
+                        .build());
+    }
```

Comment:
`error()` 메서드는 `GlobalExceptionHandler` 클래스에서만 사용하고, `GlobalExceptionHandler` 클래스를 보면 실패 응답에 대해서 받는 data는 항상 null로 처리하고 있습니다. 
그렇다면 현재 코드의 `error()` 메서드에서 data를 받을 필요가 있을까요?
```suggestion
    // 실패 응답
    public static <T> ResponseEntity<ApiResponseTemplate<T>> error(ErrorCode errorCode) {
        return ResponseEntity
                .status(errorCode.getHttpStatus())
                .body(ApiResponseTemplate.<T>builder()
                        .status(errorCode.getHttpStatus().value())
                        .success(false)
                        .message(errorCode.getMessage())
                        .build());
    }
```

Reply:
그렇네요! 수정하였습니다!